### ISSUE TITLE
Link that breaks Flash.pm

### ISSUE THREAD

**Issue by dkindlund** [dkindlund/honeyclient#108]
Specifically
http://lads.myspace.com/music/musicplayer.swf?n=aHR0cDovL211c2ljLm15c3BhY2UuY29t&t=RuFxQ/reN4Jcr/VU/CWfuEHmCNCJMjbNYvbCTe5YmK6REBM6OjXQu+NPJCZ0X1TB1YBlAWWwhOJwbee/4pEmdw==&u=LTE=&a=0&d=MTI2MTkzMDE1XjExOTA4OTIzODY=

error
{{{
2007-09-28 09:42:01  INFO [HoneyClient::Agent::worker](lib/HoneyClient/Agent.pm:956) - HoneyClient::Agent::Driver::Brow
ser::IE - Integrity Check: PASSED
HoneyClient::Agent::Driver::Browser::IE - About to return out of child thread.
HoneyClient::Agent::Driver::Browser::IE - Checking TID = 4
HoneyClient::Agent::Driver::Browser::IE - Thread NOT defined.
HoneyClient::Agent::Driver::Browser::IE - Creating a new run() child thread...
HoneyClient::Agent::Driver::Browser::IE - Thread ID = 5
Run thread(s) initialized.
2007-09-28 09:42:03  INFO [HoneyClient::Agent::worker](lib/HoneyClient/Agent.pm:902) - HoneyClient::Agent::Driver::Brow
ser::IE - Driving To Resource: http://lads.myspace.com/music/musicplayer.swf?n=aHR0cDovL211c2ljLm15c3BhY2UuY29t&t=RuFxQ/
reN4Jcr/VU/CWfuEHmCNCJMjbNYvbCTe5YmK6REBM6OjXQu+NPJCZ0X1TB1YBlAWWwhOJwbee/4pEmdw==&u=LTE=&a=0&d=MTI2MTkzMDE1XjExOTA4OTIz
ODY=
     20 [unknown (0x3A8)] perl 1936 fork: child 932 - died waiting for dll loading, errno 11
2007-09-28 09:42:10 ERROR [HoneyClient::Agent::worker](lib/HoneyClient/Agent.pm:994) - HoneyClient::Agent::Driver::Brow
ser::IE - FAULT: panic: MUTEX_LOCK (45) [util.c:2266] at lib/HoneyClient/Agent/Driver/ActiveContent/Flash.pm line 229.
Thread 5 terminated abnormally: panic: MUTEX_LOCK (45) [pad.c:1356] at /usr/lib/perl5/5.8/cygwin/Sys/Syslog.pm line 174.
}}}

And then the manager never properly suspends the VM and just sits there with 

{{{
VM Integrity Check: OK!
Cannot encode 'sources' element as 'hash'. Will be encoded as 'map' instead
Cannot encode 'value' element as 'hash'. Will be encoded as 'map' instead
VM State Table:
$VAR1 = {
          'fee5e5c7e28d4ae6a44694de89' => {
                                          'targets' => {
                                                         'lads.myspace.com' => {
                                                                                 'tcp' => [
                                                                                            80
                                                                                          ]
                                                                               }
                                                       },
                                          'sources' => {
                                                         '00:0c:29:41:36:5a' => {
                                                                                '10.0.0.182' => {
                                                                                                'tcp' => undef
                                                                                              }
                                                                              }
                                                       }
                                        }
        };

Cannot encode 'sources' element as 'hash'. Will be encoded as 'map' instead
Cannot encode 'value' element as 'hash'. Will be encoded as 'map' instead
Calling run()...
Sleeping for 2s...
Calling getState()...

}}}

**Comment by dkindlund** [comment]
Author: kindlund
Here's another link that also breaks it:

http://veoliaes.com/ExtractServlet?id=VES.TOPFLASH.015.swf

{{{
2007-10-22 14:08:22  INFO [HoneyClient::Agent::worker](lib/HoneyClient/Agent.pm:902) - HoneyClient::Agent::Driver::Brow
ser::IE - Driving To Resource: http://veoliaes.com/ExtractServlet?id=VES.TOPFLASH.015.swf
     21 [unknown (0x268)] perl 1524 fork: child 1016 - died waiting for dll loading, errno 11
2007-10-22 14:08:29 ERROR [HoneyClient::Agent::worker](lib/HoneyClient/Agent.pm:993) - HoneyClient::Agent::Driver::Browser::IE - FAULT: panic: MUTEX_LOCK (45) [util.c:2266] at lib/HoneyClient/Agent/Driver/ActiveContent/Flash.pm line 229.
Thread 4 terminated abnormally: panic: MUTEX_LOCK (45) [pad.c:1356] at /usr/lib/perl5/5.8/cygwin/Sys/Syslog.pm line 174.
}}}

In a nutshell, the Flash code is getting called in a way such that when an error occurs, the perl interpreter panics instead of failing gracefully.  We need to:
1. Fix how perl handles panics from buggy Flash parsing.
2. Fix the original bug that caused Flash to crash in the first place.

**Comment by dkindlund** [comment]
Author: xkovah
I have another link for this:
http://www.youtube.com/v/cMN6enesgPA&rel=1

**Comment by dkindlund** [comment]
Author: xkovah
and http://www.youtube.com/v/IJMqdUGZZ9s&rel=1 just for the sake of testing 2 from the same site.

**Comment by dkindlund** [comment]
Author: xkovah
I just went to all of the above links successfully, therefore this issue was probably fixed by some of the recent changes, and I will close it until we see it happen again.

**Comment by dkindlund** [comment]
Author: xkovah
bah, I accidentally tested this with the active content processing disabled.
